src/loader/implementation.ml

```diff
@@ -0,0 +1,517 @@
+#if OCAML_VERSION >= (4, 14, 0)
+
+(* open Odoc_model.Lang.Source_info *)
+
+let pos_of_loc loc = (loc.Location.loc_start.pos_cnum, loc.loc_end.pos_cnum)
+
+type annotations =
+  | LocalDefinition of Ident.t
+  | LocalValue of Ident.t
+  | DefJmp of Shape.Uid.t
+
+module Analysis = struct
+  let ( @ ) = List.rev_append
```

Comment:
I don't like the redefinition of `@` to be semantically different from the usual `@`! It's confusing.

Also I'm not sure that we care too much about the ordering of things do we? There are a bunch of `List.rev`s below that can be removed if we don't.

Reply:
I guess we can turn the "lift" into a "fold" and carry an accumulator around, that would be the best option.

Yes, the order does not matter.

Reply:
Turned the traverse into a fold with an accumulator in 9361f0a. Hopefully it's better like that, with less list concatenations!